src/httpd_jsonapi.c

```diff
@@ -2903,6 +2909,44 @@ jsonapi_reply_library_playlist_tracks(struct httpd_request *hreq)
   return HTTP_OK;
 }
 
+static int
+jsonapi_reply_library_playlist_from_queue(struct httpd_request *hreq)
+{
+  const char *param;
+  int ret = 0;
+  char buf[PATH_MAX+15];
+  char *plsname = NULL;
+
+  if ( (param = evhttp_find_header(hreq->query, "name")) == NULL)
+    {
+      DPRINTF(E_LOG, L_WEB, "Invalid argument, missing 'name'\n");
+      return HTTP_BADREQUEST;
+    }
+
+  if (default_pl_dir == NULL || !allow_modifying_stored_playlists)
+    return HTTP_NOTIMPLEMENTED; 
+
+  if (default_pl_dir && access(default_pl_dir, W_OK) < 0)
+    {
+      DPRINTF(E_LOG, L_WEB, "invalid playlist save directory=%s\n", default_pl_dir? default_pl_dir : "NULL");
```

Comment:
`default_pl_dir` cannot be NULL at this point and http return code should be 403